src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,38 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final String USER_DETAILS = "Details for ";
+    private static final Random random = new Random();
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> getUserData(userId), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            Thread.sleep(random.nextInt(1000));
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        }
+        UserData cachedUserData = cache.get(userId);
+        if (cachedUserData != null) {
+            return cachedUserData;
+        }
+        UserData userData = new UserData(userId, USER_DETAILS + userId);
```

Comment:
The creation of a new UserData object and subsequent update to the cache is not atomic. If multiple threads call `getUserData` with the same userId at the same time, it could lead to the creation of multiple UserData instances for the same userId. Consider using `cache.computeIfAbsent` to handle this atomically.